src/Data/Unfoldable.purs

```diff
@@ -88,6 +89,13 @@ none = unfoldr (const Nothing) unit
 singleton :: forall f a. Unfoldable f => a -> f a
 singleton = replicate 1
 
+-- | Produce a Lazy Unfoldable structure representing the natural numbers.
+nats :: forall f a. Unfoldable f => Semiring a => Lazy (f a) => f a
+nats = defer \_ -> unfoldr increment zero
```

Comment:
I don't think you need `defer` here, since any lazy structure should use it anyway. Did you try this on lazy lists?

Reply:
Yes, you are right. Will fix.